Automattic/wp-calypso
# WP data stores: fix store persistence misconfiguration and add selectors for siteTitle, logo and description

### Explanation

* For a couple of years, we had a sneaky bug in our stores persistence configuration. We had [a factory](https://github.com/Automattic/wp-calypso/blob/trunk/packages/data-stores/src/persistence-config-factory/index.ts#L11) that creates `persistenceConfig` for every store we register. And this factory accepts the store key to use it as `localStorage` key for persisting that store.
* The problem is that while WP stores persistence is indeed configurable, the configuration applies **globally**. And has nothing to do with the particular store. Meaning, we don't know which storage key will be used at a time because we kept reconfiguring the **global** persistence configuration every time we register a store ([one](https://github.com/Automattic/wp-calypso/blob/trunk/packages/data-stores/src/help-center/index.ts#L14), [two](https://github.com/Automattic/wp-calypso/blob/trunk/packages/data-stores/src/onboard/index.ts#L19)), and who knows which configuration took control? The first one? The last one? That was undefined behavior and only surfaced a bug today.
* This bug will only surface if two or more stores that use persistence are registered, then the session is interrupted, then the session resumes, and then some stores will use the wrong config that was set by other stores.

### The fix
* I fix this by unifying the key for all persistence in all stores. WP Data stores are meant to be used like this, and they fragment the stores based on the store key anyway! We don't need to fragment ourselves. 

### Testing Instructions

This affects a few places, let's do a quick test for all of them. For starters, please check this branch out and run `yarn start` and in ETK run `yarn dev --sync`.

#### help-center score
1. Go to a sandboxed site editor.
2. Open the help center.
3. Reach the contact form and fill it.
4. Refresh the page and repeat.
5. The form should be full.

#### The onboard store
1. Go through a tailored flow (make sure to upload a logo).
2. The site should have title, description, and logo.

#### The reader store
1. It isn't used anywhere, skip

#### The site store
1. You can go though /setup?siteSlug=YOUR_SITE.
2. The site setup flow uses the site store extensively. 
3. If it works, the store is good.
